src/Compilers/CSharp/Portable/CommandLine/CSharpCompiler.cs

```diff
@@ -81,6 +81,21 @@ protected CSharpCompiler(CSharpCommandLineParser parser, string? responseFile, s
             var normalizedFilePaths = new string?[sourceFiles.Length];
             var diagnosticBag = DiagnosticBag.GetInstance();
 
+            // <Metalama>
+            string? langVersionBeforeOverride = null;
+            globalConfigOptions.AnalyzerOptions?.TryGetValue("build_property.MetalamaLangVersionBeforeOverride", out langVersionBeforeOverride);
+
+            if (!string.IsNullOrWhiteSpace(langVersionBeforeOverride))
+            {
+                diagnosticBag.Add(
```

Comment:
You can emit this warning in MSBuild, so you don't need any C# code or compiler-visible property.

https://learn.microsoft.com/en-us/visualstudio/msbuild/warning-task?view=vs-2022

Reply:
Thanks, yeah, that's simpler.